src/fiatconnect/saga.ts

```diff
@@ -893,97 +893,120 @@ export function* handleFetchFiatConnectProviders() {
+  const context = newTransactionContext(TAG, 'Send crypto to provider for transfer out')
 
-      const context = newTransactionContext(TAG, 'Send crypto to provider for transfer out')
+  const { error, receipt }: { receipt: CeloTxReceipt; error: any } = yield call(
+    buildAndSendPayment,
+    context,
+    transferAddress,
+    new BigNumber(fiatConnectQuote.getCryptoAmount()),
+    tokenInfo.address,
+    '',
+    feeInfo!
+  )
+  if (error) {
+    ValoraAnalytics.track(FiatExchangeEvents.cico_fc_transfer_tx_error, {
+      flow: CICOFlow.CashOut,
+      error: error.message,
+      transferAddress: transferAddress,
+      provider: fiatConnectQuote.getProviderId(),
+    })
+    throw error
+  }
+  Logger.info(TAG, 'Completed transfer out transaction..')
+  return receipt.transactionHash
+}
 
-      const { receipt, error }: { receipt: CeloTxReceipt; error: any } = yield call(
-        buildAndSendPayment,
-        context,
-        transferResult.transferAddress,
-        new BigNumber(fiatConnectQuote.getCryptoAmount()),
-        tokenInfo.address,
-        '',
-        feeInfo!
-      )
+export function* handleCreateFiatConnectTransfer(
+  action: ReturnType<typeof createFiatConnectTransfer>
+) {
+  const { flow, fiatConnectQuote } = action.payload
+  const quoteId = fiatConnectQuote.getQuoteId()
+  let transactionHash: string | null = null
+  try {
+    const { transferAddress }: TransferResponse = yield call(_initiateTransfer, action)
 
-      if (error) {
-        ValoraAnalytics.track(FiatExchangeEvents.cico_fc_transfer_tx_error, {
-          flow,
-          error: error.message,
-          transferAddress: transferResult.transferAddress,
-          provider: fiatConnectQuote.getProviderId(),
-        })
-        throw error
-      }
+    if (flow === CICOFlow.CashOut) {
+      transactionHash = yield call(_initiateTransaction, {
+        transferAddress,
+        fiatConnectQuote,
+      })
+    }
 
-      ValoraAnalytics.track(FiatExchangeEvents.cico_fc_transfer_success, {
-        txHash: receipt.transactionHash,
-        transferAddress: transferResult.transferAddress,
-        provider: fiatConnectQuote.getProviderId(),
+    ValoraAnalytics.track(FiatExchangeEvents.cico_fc_transfer_success, {
+      txHash: transactionHash,
+      transferAddress: transferAddress,
+      provider: fiatConnectQuote.getProviderId(),
+      flow,
+    })
+    yield put(
+      createFiatConnectTransferCompleted({
         flow,
+        quoteId,
+        txHash: transactionHash,
       })
-
-      yield put(
-        createFiatConnectTransferCompleted({
-          flow,
-          quoteId,
-          txHash: receipt.transactionHash,
-        })
-      )
-    } catch (err) {
-      Logger.error(TAG, 'Transfer out failed..', err)
-      yield put(createFiatConnectTransferFailed({ flow, quoteId }))
-    }
-  } else {
-    throw new Error('not implemented')
+    )
+  } catch (err) {
+    Logger.error(TAG, `Transfer for ${flow} failed..`, err)
```

Comment:
Should we log an analytics event here on error? There are code paths in the callees of this method that could throw but don't send an analytics event.

Reply:
mmm, good catch. I'll move the Analytics error log here.

Reply:
what are the other paths that could throw? We'd end up logging multiple events in some cases right? If for example `transfer_api_error` or `tx_error` is logged.

Reply:
A runtime error in `buildAndSendPayment` or maybe an error setting up `fiatConnectClient` ?

I thought it definitely isn't ideal that we log two events in some (most) cases. But since this is such an important step in the flow I thought it was acceptable so that we cover all possible cases to be able to help us debug.